Add missing include needed for tests (and possibly consumers) with gcc11.

Fixed upstream in:
https://github.com/ThePhD/sol2/commit/e5e6466e09b632677d24a8f204d6a0ea0a8862b1
--- a/include/sol/stack_core.hpp
+++ b/include/sol/stack_core.hpp
@@ -42,2 +42,3 @@
 #include <string>
+#include <limits>
 #include <algorithm>
